Sources/NIO/ChannelOption.swift

```diff
@@ -225,9 +255,21 @@ extension ChannelOptions {
 
 /// Provides `ChannelOption`s to be used with a `Channel`, `Bootstrap` or `ServerBootstrap`.
 public struct ChannelOptions {
+    #if !os(Windows)
+        @available(*, deprecated, renamed: "ChannelOptions.socketOption(_:)")
+        public static let socket = { (level: SocketOptionLevel, name: SocketOptionName) -> Types.SocketOption in
+            .init(level: NIOBSDSocket.OptionLevel(rawValue: CInt(level)), name: NIOBSDSocket.Option(rawValue: CInt(name)))
+        }
+    #endif
+
+    /// - seealso: `SocketOption`.
+    public static let socketOption = { (name: NIOBSDSocket.Option) -> Types.SocketOption in
+        .init(level: .socket, name: name)
+    }
+
     /// - seealso: `SocketOption`.
-    public static let socket = { (level: SocketOptionLevel, name: SocketOptionName) -> Types.SocketOption in
-        .init(level: level, name: name)
+    public static let tcp = { (name: NIOBSDSocket.Option) -> Types.SocketOption in
```

Comment:
I wonder if this should be `tcpOption` for consistency with the new `socketOption`.

Reply:
Sadly I have to agree with you (but I can do so begrudgingly right?).  I really like `tcp`, but it is nicer to have the uniform `tcpOption` and `socketOption`.